sklearn_numba_dpex/common/kernels.py

```diff
@@ -156,21 +166,107 @@ def make_sum_reduction_1d_kernel(size, work_group_size, device, dtype):
     two_long = np.int64(2)
     one_idx = np.int64(1)
 
+    is_1d = size1 is None
+    if is_1d:
+        sum_axis_size = size0
+        n_rows = np.int64(1)
+        local_data_size = work_group_size
+
+        @dpex.func
+        def set_col_to_zero(array, i):
+            array[i] = zero
+
+        @dpex.func
+        def copy_col(from_array, from_col, to_array, to_col):
+            to_array[to_col] = from_array[from_col]
+
+        @dpex.func
+        def coalesce_cols(
```

Comment:
Since the coalescences are additions only, would it make sense to use `add_*_cols_*` instead of `coalesce_*_cols_*`?

Reply:
I liked `coalesce` here because it stresses the fact that the kernel could be abstracted to and extended to other functions that can be accumulated (like min, max, etc), just by replacing the "sum" op

Reply:
I see.

I think we started using `reduce` and `reduction` as terms commonly. Is this semantic different?

For now, I am +0 regarding creating private abstractions for coalescences (or reductions) at this level mainly because YAGNI and because private renaming are cheap.

Reply:
I only intended to hint into the possibility of this abstraction, not implement it now. But I agree with your point I'll just revert to naming to `add_*_cols_*` and explicit the hint in a comment.